web/src/app/profile/approved-researcher/components/ApprovedResearcher.tsx

```diff
@@ -4,18 +4,39 @@ import { useAuth } from "../../../hooks/useAuth";
 import LoginFallback from "@/components/ui/LoginFallback";
 import ApprovedResearcherForm from "./ApprovedResearcherForm";
 import AgreementText from "./AgreementText";
+import { Agreement, getAgreementsApprovedResearcher } from "@/openapi";
+import { useEffect, useState } from "react";
 
 export default function ApprovedResearcher() {
-  const { loading, isAuthed } = useAuth();
+  const { isAuthed } = useAuth();
+  const [agreement, setAgreement] = useState<Agreement | null>(null);
```

Comment:
You may need:

` if (loading) return null;`

In order to block further content rendering until the auth is established. Otherwise, you might see a brief flash of the Login button.